### ISSUE TITLE
<Transition> element exits without animating when content is changed with setState

### ISSUE THREAD

**Issue by jlacson123** [react-spring/react-spring#859]
## 🤓 Question
I am wondering if this is supported or not, or maybe I am just doing it the wrong way. But I have content which is supposed to unmount on click as it also gets removed from state. Animation works perfectly when entering but upon exit it just disappears without animating. 
I can't post the actual code so here's a snippet that's similar
`<Transition items={clicked} from={{ opacity: 0 }} enter={{ opacity: 1 }} leave={{ opacity: 0 }}>` 
`{item => (item ? props => <div style={{ ...props, overflow: 'hidden' }}>{this.state.content}</div> : '')}`
`</Transition>`

Sorry I hope it's not a dumb question, there's nothing much about this in the docs :(

**Comment by aleclarson** [comment]
You should use `animated.div` instead of a plain `div`.

```ts
import { animated } from 'react-spring'
```

**Comment by jelacs** [comment]
@aleclarson I don't know why this was closed, unfortunately the solution given did not fix my problem.
I'm trying to create a list of expandable and collapsible items whose content comes from a single property in component state.
My problem is on the exit animation, the content just disappears (if expanded item is clicked to collapse) or is updated with the new state (if another item is clicked). I tried to replicate the issue [here](https://codesandbox.io/s/react-spring-state-change-8zztf).
